### ISSUE TITLE
How about "app spread" in the context menu?

### ISSUE THREAD

**Issue by HubKing** [micheleg/dash-to-dock#1951]
Currently, there are not many items in the context menu, only "unpin", "quit", etc. I think having a menu for that "app spread" would not hurt. There currently is "All Windows", but they are very small.

Or maybe, in the Behaviour settings, adding an entry for just "app spread", not "focus or app spread", so that the user could map that entry to some click action.

In short, I would like to have an easier way to invoke app spread.

**Comment by vanvugt** [comment]
Isn't tapping the Super (Windows) key an easy way to invoke the app spread? Or are you asking for something different?

**Comment by HubKing** [comment]
No, app spread and Super (Overview) are different. The former shows only the windows of that application, and the latter shows all open windows.

**Comment by 3v1n0** [comment]
I think it would also make some sense to have a shortcut to show the app spread for windows, indeed we could add a further behavior too.

**Comment by nitro122** [comment]
Will this be implemented?

**Comment by nitro122** [comment]
it would be nice if someone could implement it as a gesture for touchpads

**Comment by neko-kai** [comment]
@nitro122 I did, in a fork of `touchpad-gesture-customization` extension https://github.com/7mind/touchpad-gesture-customization-app-expose

**Comment by nitro122** [comment]
> [@nitro122](https://github.com/nitro122) I did, in a fork of `touchpad-gesture-customization` extension https://github.com/7mind/touchpad-gesture-customization-app-expose

hey @neko-kai  thank you for you reply. I installed your extension in the link you provided but could not find where option to control app spread. If you could please explain how to set it up I would be very grateful :).

**Comment by neko-kai** [comment]
@nitro122 You need to get to GNOME Extensions app (`gnome-extensions-app`), go to Touchpad Gesture Customization extension's settings, set 3 or 4 finger vertical swipe to Overview Navigation and in 'Misc' section, set `Overview navigation states` to `Application overview on swipe down` (it's cropped for me to 'Application overview on...', not sure if that's a font size issue), screenshots:

<img width="1378" height="1136" alt="Image" src="https://github.com/user-attachments/assets/1b65e0df-8349-4f3e-8e61-e67daceeae18" />

<img width="1378" height="1136" alt="Image" src="https://github.com/user-attachments/assets/2787082d-fb77-42cc-9c1e-a90255680855" />

Then swiping up will bring up overview of all apps as usual, and swiping down, overview of current app only.

**Comment by nitro122** [comment]
<img width="632" height="291" alt="Image" src="https://github.com/user-attachments/assets/e3ba8f07-38e1-4d4d-aa59-9e0cfc67074e" />

**Comment by neko-kai** [comment]
@nitro122 That means you installed the non-fork version, probably by following the link to gnome store where the fork is _not_ published. You need to uninstall it - via UI or however, and for good measure via command-line: `gnome-extensions uninstall touchpad-gesture-customization@coooolapps.com`, then follow the manual installation instructions in fork https://github.com/7mind/touchpad-gesture-customization-app-expose/blob/main/README.md#manually

**Comment by nitro122** [comment]
thankyou I followed the instructions on your guide and  got the forked version installed. But it doesnt seem to work

<img width="1382" height="684" alt="Image" src="https://github.com/user-attachments/assets/fc5e0ef3-ffb6-4a13-86d9-82cefd00be15" />

**Comment by neko-kai** [comment]
@nitro122 You need to make sure the extension is enabled and active in `gnome-extensions info touchpad-gesture-customization@coooolapps.com`, also changes to extensions in GNOME require logging out and logging back in again.

**Comment by nitro122** [comment]
> [@nitro122](https://github.com/nitro122) You need to make sure the extension is enabled and active in `gnome-extensions info touchpad-gesture-customization@coooolapps.com`, also changes to extensions in GNOME require logging out and logging back in again.

I enabled it in extension as seen the picture. Or do you mean it in another way. I logged -out and also restarted

**Comment by neko-kai** [comment]
@nitro122 What is the error message? Are you on GNOME 49 or 50?

**Comment by nitro122** [comment]
> [@nitro122](https://github.com/nitro122) What is the error message? Are you on GNOME 49 or 50?

no error

<img width="685" height="220" alt="Image" src="https://github.com/user-attachments/assets/4b14c058-3b30-4bdb-930d-7993d94470ef" />

gnome = 48 
Debian = Trixie

**Comment by neko-kai** [comment]
@nitro122 The extension is for gnome 49 and gnome 50 only (see https://github.com/7mind/touchpad-gesture-customization-app-expose/blob/main/metadata.json) you'd need to backport the fork's changes over `48` branch of upstream extension (https://github.com/HieuTNg/touchpad-gesture-customization/tree/48) or update GNOME.

**Comment by nitro122** [comment]
> [@nitro122](https://github.com/nitro122) The extension is for gnome 49 and gnome 50 only (see https://github.com/7mind/touchpad-gesture-customization-app-expose/blob/main/metadata.json) you'd need to backport the fork's changes over `48` branch of upstream extension (https://github.com/HieuTNg/touchpad-gesture-customization/tree/48) or update GNOME.

Thankyou for your insight but I think i am not that equipped to do this. :(

**Comment by nitro122** [comment]
> [@nitro122](https://github.com/nitro122) The extension is for gnome 49 and gnome 50 only (see https://github.com/7mind/touchpad-gesture-customization-app-expose/blob/main/metadata.json) you'd need to backport the fork's changes over `48` branch of upstream extension (https://github.com/HieuTNg/touchpad-gesture-customization/tree/48) or update GNOME.

Is there a guide on how to use this extension with Gnome 48? I really want to use this feature. But I don't know how to do this. My distro is Debian, so I cannot update to Gnome 49 until 2028.

**Comment by neko-kai** [comment]
@nitro122 I ported it by LLM to GNOME 48, you may try again from fresh main (no idea whether it'll work though)

**Comment by nitro122** [comment]
thx but unfortunately not.
This is what i get :

<img width="511" height="266" alt="Image" src="https://github.com/user-attachments/assets/18563ebd-773d-4bca-8d3a-faa8543bcf16" />

**Comment by neko-kai** [comment]
@nitro122 Make sure you fully uninstall the old extension version first, before installing new version, and that uninstallation succeeded (there were no errors):
```
gnome-extensions uninstall touchpad-gesture-customization@coooolapps.com
```
And that you clone the most recent repository version from scratch and follow the installation instructions (https://github.com/7mind/touchpad-gesture-customization-app-expose/blob/main/README.md#manually):

```
cd /tmp
git clone https://github.com/7mind/touchpad-gesture-customization-app-expose.git
cd touchpad-gesture-customization-app-expose
npm install
npm run update
```

Log out of GNOME and log back in, enable extension via command line:
```
gnome-extensions enable touchpad-gesture-customization@coooolapps.com
```
